src/redux/sagas/actions/manageExistingSafes.ts

```diff
@@ -0,0 +1,273 @@
+    // const ipfsWithFallback = getContext(ContextModule.IPFSWithFallback);
+
+    txChannel = yield call(getTxChannel, metaId);
+
+    const batchKey = !isRemovingSafes
+      ? 'addExistingSafe'
+      : 'removeExistingSafes';
+
+    const {
+      manageExistingSafesAction: manageExistingSafes,
+      // annotateManageExistingSafesAction: annotateManageExistingSafes,
+    } = yield createTransactionChannels(metaId, [
+      'manageExistingSafesAction',
+      // 'annotateManageExistingSafesAction',
+    ]);
+
+    const createGroupTransaction = ({ id, index }, config) =>
+      fork(createTransaction, id, {
+        ...config,
+        group: {
+          key: batchKey,
+          id: metaId,
+          index,
+        },
+      });
+
+    yield createGroupTransaction(manageExistingSafes, {
+      context: ClientType.ColonyClient,
+      methodName: 'editColony',
+      identifier: colonyAddress,
+      params: [],
+      ready: false,
+    });
+
+    // if (annotationMessage) {
+    //   yield createGroupTransaction(annotateManageExistingSafes, {
+    //     context: ClientType.ColonyClient,
+    //     methodName: 'annotateTransaction',
+    //     identifier: colonyAddress,
+    //     params: [],
+    //     ready: false,
+    //   });
+    // }
+
+    yield takeFrom(
+      manageExistingSafes.channel,
+      ActionTypes.TRANSACTION_CREATED,
+    );
+
+    // if (annotationMessage) {
+    //   yield takeFrom(
+    //     annotateManageExistingSafes.channel,
+    //     ActionTypes.TRANSACTION_CREATED,
+    //   );
+    // }
+
+    yield put(transactionPending(manageExistingSafes.id));
+
+    /*
+     * Fetch colony data from the subgraph
+     * And destructure the metadata hash.
+     */
+
+    // const {
+    //   data: {
+    //     colony: { metadata: currentMetadataIPFSHash },
+    //   },
+    // } = yield apolloClient.query<
+    //   SubgraphColonyQuery,
+    //   SubgraphColonyQueryVariables
+    // >({
+    //   query: SubgraphColonyDocument,
+    //   variables: {
+    //     address: colonyAddress.toLowerCase(),
+    //   },
+    //   fetchPolicy: 'network-only',
+    // });
+
+    // const currentMetadata = yield call(
+    //   ipfsWithFallback.getString,
+    //   currentMetadataIPFSHash,
+    // );
+
+    // if (!currentMetadata) {
+    //   throw new Error(
+    //     `There was an error while fetching the current colony metadata. Please try again later.`,
+    //   );
+    // }
+
+    // const parsedColonyMetadata = JSON.parse(currentMetadata);
+    // const metadataVersion = getEventMetadataVersion(currentMetadata);
+    // const currentColonyMetadata =
+    //   metadataVersion === 1
+    //     ? { ...parsedColonyMetadata }
+    //     : parsedColonyMetadata.data;
+
+    let updatedColonySafes: Safe[];
+
+    const colonySafes =
+      colony.metadata?.safes
+        ?.filter(notNull)
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        .map(({ __typename, ...rest }) => rest) || [];
```

Comment:
There's an `excludeTypenameKey` util in `~utils/object` that does that without needing to disable eslint rules